### PR TITLE
Adding continuous deployent for Dawg

### PR DESCRIPTION
**Pull request by kpears201**

### CODE REVIEW

File: automation/src/main/resources/puppet/manifests/dawg.pp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            packaging => 'war',
+            ensure => 'present',
+   }
+
+   file { '/etc/dawg':
+        ensure => 'directory',
+        owner => 'tomcat',
+   }
+
+   config_file::config_file { 'dawg.ini':
```
Comment (kpears201) [inline automation/src/main/resources/puppet/manifests/dawg.pp:72]: Note: I am using this puppet module I wrote awhile back. I would like to opensource that module too.

File: libraries/dawg-client/pom.xml [hunk 2]
```diff
@@ -8,6 +8,7 @@
     </parent>
 
     <artifactId>dawg-client</artifactId>
+    <version>2.0.2-SNAPSHOT</version>
```
Comment (kpears201) [inline libraries/dawg-client/pom.xml:11]: I changed all the versions to be explicitly set rather than inferred. We use the maven set versions plugin in continuous deployment and it needs the versions to be explicitly set. These values of a version should be handled automatically like in the maven-release-plugin or maven set versions plugin anyway so it doesn't require them to all be changed one by one manually.

File: libraries/pom.xml [hunk 3]
```diff
@@ -8,13 +8,13 @@
...
 
     <artifactId>libraries</artifactId>
+    <version>2.0.2-SNAPSHOT</version>
     <packaging>pom</packaging>
 
     <name>(Comcast) ${project.artifactId}</name>
     <description>DAWG library</description>
 
     <modules>
-        <module>dawg-config</module>
```
Comment (kpears201) [inline libraries/pom.xml:17]: I removed this... Seems dawg-config is literally just an example config file. We can keep it in the repo for reference but I am not so sure it needs get put into an RPM and be part of the build and deployment.

**Review by kpears201 — COMMENTED** [review]
This is to support automated continuous deployment. It has the puppet files used to set up tomcat, deploy wars and create the config file.

File: automation/src/main/resources/puppet/manifests/dawg.pp (hunk 1 above)
Comment (dimaj) [reply automation/src/main/resources/puppet/manifests/dawg.pp:72]: how will this work now if that module is still internal?

File: automation/src/main/resources/puppet/manifests/dawg.pp [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    file { '/etc/init.d/tomcat':
+        content => hiera('tomcat.init'),
+        mode => '0755',
+        require => Tomcat::Install['/opt/tomcat'],
+    }
+
+    exec {"genunit-tomcat":
+        path => '/sbin:/bin:/usr/sbin:/usr/bin',
+        onlyif => 'test -f /bin/systemctl',
```
Comment (dimaj) [inline automation/src/main/resources/puppet/manifests/dawg.pp:19]: i don't think you have to have `systemctl`. You can always go back to `service tomcat start`

File: pom.xml [hunk 5]
```diff
@@ -516,28 +510,6 @@
                 <artifactId>httpcore</artifactId>
                 <version>${version.apache.http}</version>
             </dependency>
-
```
Comment (dimaj) [inline pom.xml:519]: why did you remove these? they're dependency management so, they won't be brought it unless asked for. By having them here, we have a single point of version management, no?
Speaking of version managements, I think we should move all dependencies into the `dependencyManagement` section and have individual modules bring in only what they need

**Review by dimaj — COMMENTED** [review]
some general questions

File: automation/src/main/resources/puppet/manifests/dawg.pp [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+	class {'nexus':
+	   url => $nexurl
+    }
+
+    $version = hiera('dawg.version')
+    $repo = hiera('nexus.repo')
+    $groupId = "com.comcast.video.dawg"
+
+    $dsArtifactId = "dawg-show"
```
Comment (trentontrees) [inline automation/src/main/resources/puppet/manifests/dawg.pp:40]: Why install all of the dawg components via war files instead of the rpms?

File: automation/src/main/resources/puppet/manifests/dawg.pp (hunk 1 above)
Comment (kpears201) [reply automation/src/main/resources/puppet/manifests/dawg.pp:72]: it will only work internally for now. The puppet module gets installed on the machine and then it can be referenced. I suppose I can just use an erb template for now. But its pretty likely this automation won't be needed by others very soon.

File: automation/src/main/resources/puppet/manifests/dawg.pp (hunk 4 above)
Comment (kpears201) [reply automation/src/main/resources/puppet/manifests/dawg.pp:19]: I believe you need it for centos7 now. The unit file needs to be generated for the init script.

File: pom.xml (hunk 5 above)
Comment (kpears201) [reply pom.xml:519]: We really shouldn't be using dependency management for artifacts that are a part of this project. The version will always be the project version. As mentioned before, the versions need to be explicitly set for our project version in order to use the maven set versions plugin, and these versions don't need to be changed manually but as part of a release plugin or something similar

File: libraries/pom.xml (hunk 3 above)
Comment (trentontrees) [reply libraries/pom.xml:17]: dawg-config builds an rpm which contains the default information.  This rpm is required for installing the dawg -show/-house/-pound rpms.

File: automation/src/main/resources/puppet/manifests/dawg.pp (hunk 6 above)
Comment (dimaj) [reply automation/src/main/resources/puppet/manifests/dawg.pp:40]: because we can then install it on any system (linux, win, mac)

Comment (kpears201) [reply automation/src/main/resources/puppet/manifests/dawg.pp:40]: easier.
General consensus is yum is the worst thing to have ever existed.

File: automation/src/main/resources/puppet/manifests/dawg.pp (hunk 1 above)
Comment (dimaj) [reply automation/src/main/resources/puppet/manifests/dawg.pp:72]: ok

File: automation/src/main/resources/puppet/manifests/dawg.pp (hunk 6 above)
Comment (kpears201) [reply automation/src/main/resources/puppet/manifests/dawg.pp:40]: Also, using an RPM then requires it to be rebuilt if you want to install it into a different web container like jboss or something.

File: libraries/pom.xml (hunk 3 above)
Comment (kpears201) [reply libraries/pom.xml:17]: But the default information is pretty useless, it just sets things to localhost.

File: automation/src/main/resources/puppet/manifests/dawg.pp (hunk 4 above)
Comment (dimaj) [reply automation/src/main/resources/puppet/manifests/dawg.pp:19]: https://github.com/puppetlabs/puppetlabs-tomcat/blob/master/manifests/service.pp#L76

Comment (kpears201) [reply automation/src/main/resources/puppet/manifests/dawg.pp:19]: Yeah I tried it, doesn't really work.

Comment (kpears201) [reply automation/src/main/resources/puppet/manifests/dawg.pp:19]: For some reason it was starting up two exact processes, one run by root, one run by tomcat user. And then the shutdown command didn't work.

This is easier.

Comment (dimaj) [reply automation/src/main/resources/puppet/manifests/dawg.pp:19]: ok. then you should ensure that `systemd` is installed

Comment (kpears201) [reply automation/src/main/resources/puppet/manifests/dawg.pp:19]: It comes in the base image for centos 7+
And there is a test before calling systemctl to verify that it is installed. If this is centos6, it will skip the systemctl command and just work with the init script.

**Comment by dimaj** [comment]
ok. cool


Regards,

dimaj <http://blog.dimaj.net>

File: libraries/pom.xml (hunk 3 above)
Comment (trentontrees) [reply libraries/pom.xml:17]: still needed as the other rpms depend on this

Comment (kpears201) [reply libraries/pom.xml:17]: depend as in, like an RPM dependency?

Comment (trentontrees) [reply libraries/pom.xml:17]: yes

**Comment by kpears201** [comment]
Added back in dawg-config @trentontrees
